forte/data/data_store.py

```diff
@@ -638,7 +690,78 @@ def add_annotation_raw(
         # A reference to the entry should be store in both self.__elements and
         # self.__tid_ref_dict.
         entry = self._new_annotation(type_name, begin, end, tid)
-        if not allow_duplicate and type_name in self.__elements:
+        if not allow_duplicate:
+            tid_search_result = self._get_existing_entry_tid(entry)
+            # if found existing entry
+            if tid_search_result != -1:
+                return tid_search_result
+
+        return self._add_entry_raw(Annotation, type_name, entry)
+
+    def add_audio_annotation_raw(
+        self,
+        type_name: str,
+        begin: int,
+        end: int,
+        tid: Optional[int] = None,
+        allow_duplicate=True,
+    ) -> int:
+
+        r"""This function adds an audio annotation entry with ``begin`` and
+        ``end``
+        indices to current data store object. Returns the ``tid`` for the
+        inserted entry.
+
+        Args:
+            type_name: The fully qualified type name of the new AudioAnnotation.
+            begin: Begin index of the entry.
+            end: End index of the entry.
+            tid: ``tid`` of the Annotation entry that is being added.
+                It's optional, and it will be
+                auto-assigned if not given.
+            allow_duplicate: Whether we allow duplicate in the DataStore. When
+                it's set to False, the function will return the ``tid`` of
+                existing entry if a duplicate is found. Default value is True.
+
+        Returns:
+            ``tid`` of the entry.
+        """
+        # We should create the `entry data` with the format
+        # [begin, end, tid, type_id, None, ...].
+        # A helper function _new_annotation() can be used to generate a
+        # annotation type entry data with default fields.
+        # A reference to the entry should be store in both self.__elements and
+        # self.__tid_ref_dict.
+        entry = self._new_audio_annotation(type_name, begin, end, tid)
+
+        if not allow_duplicate:
+            tid_search_result = self._get_existing_entry_tid(entry)
+            # if found existing entry
+            if tid_search_result != -1:
+                return tid_search_result
+        return self._add_entry_raw(AudioAnnotation, type_name, entry)
+
+    def _get_existing_entry_tid(self, entry: List[Any]):
+        r"""
+        This function searches for tid for existing entry tid. It return tid
```

Comment:
Should document that we search for `Annotation`-like data, and provide that which types can be searched here (and based on what to search)

Reply:
I think we might need to support non-Annotation like data in the future. So I am raising errors with messages to inform users.

Reply:
actually, we probably won't do that, the purpose of this function is only searching duplicate in position (e.g begin and end). but this is not quite important